stonesoup/sensor/radar/radar.py

```diff
@@ -0,0 +1,495 @@
+            ":math:`x,y,z` axis. i.e Roll, Pitch and Yaw.")
+    translation_offset = Property(
+        StateVector, default=StateVector([0, 0, 0]),
+        doc="The radar position in 3D Cartesian space.[x,y,z]")
+    mapping = Property(
+        np.array, default=[0, 1, 2],
+        doc="Mapping between or positions and state "
+            "dimensions. [x,y,z]")
+
+    measurement_model = Property(
+        MeasurementModel, default=None,
+        doc="The Measurement model used to generate "
+            "measurements.")
+
+    beam_shape = Property(
+        BeamShape,
+        doc="Object describing the shape of the beam.")
+
+    beam_transition_model = Property(
+        BeamTransitionModel,
+        doc="Object describing the "
+            "movement of the beam in azimuth and "
+            "elevation from the perspective of "
+            "the radar.")
+    # SNR variables
+    number_pulses = Property(
+        int, default=1,
+        doc="The number of pulses in the"
+            " radar burst.")
+    duty_cycle = Property(
+        float,
+        doc="Duty cycle is the fraction of the time "
+            "the radar it transmitting.")
+    band_width = Property(
+        float, doc="Bandwidth of the receiver in hertz.")
+    receiver_noise = Property(
+        float, doc="Noise figure of the radar in decibels.")
+    frequency = Property(
+        float, doc="Transmitted frequency in hertz.")
+    antenna_gain = Property(
+        float, doc="Total Antenna gain in decibels.")
+    beam_width = Property(
+        float, doc="Radar beam width in radians.")
+    loss = Property(
+        float, default=0, doc="Loss in decibels.")
+
+    swerling_on = Property(
+        bool, default=False,
+        doc="Is the Swerling 1 case used. If True the RCS"
+            " of the target will change for each timestep. "
+            "The random RCS follows the probability "
+            "distribution of the Swerling 1 case.")
+    rcs = Property(
+        float, default=None,
+        doc="The radar cross section of targets in meters squared.")
+
+    probability_false_alarm = Property(
+        Probability, default=1e-6,
+        doc="Probability of false alarm used in the North's approximation")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def _snr_constant(self):
+        temp = 290  # noise reference temperature (room temperature kelvin)
+        # convert from dB
+        noise_figure = 10 ** (self.receiver_noise / 10)
+        loss = 10 ** (self.loss / 10)
+        # calculate part of snr that is independent of:
+        #   rcs, transmitted power, gain and range
+        return (const.c ** 2 * self.number_pulses * self.duty_cycle) / \
+               (64 * np.pi ** 3 * const.k * temp * self.band_width *
+                noise_figure * self.frequency ** 2 * loss)
+
+    def measure(**kwargs):
+        raise NotImplementedError
```

Comment:
Shouldn't this be implemented, replacing gen_measurement?

Reply:
Yes, I think so, and it is a future set of changes. @jjosborne-dstl Do you have any reason not to do this?

Reply:
Yes this can change. Function name was from old versions of the other radar sensors

Reply:
This is now fixed in the latest push.